global_costmap:
  global_costmap:
    ros__parameters:
      update_frequency: 3.0
      publish_frequency: 1.0
      global_frame: map
      robot_base_frame: base_link
      use_sim_time: False
      robot_radius: 0.5
      resolution: 0.05
      rolling_window: false
      track_unknown_space: true
      footprint: "[[-0.4, -0.35], [-0.4, 0.35], [0.4, 0.35], [0.4, -0.35]]"
      plugins: ["static_layer", "depth_camera", "inflation_layer"]
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        map_subscribe_transient_local: True
      depth_camera:
        plugin: "nav2_costmap_2d::DepthCameraObstacleLayer"
        use_global_frame_to_mark: True
        forced_clearing_distance: 0.1
        ec_seg_distance: 0.2
        ec_cluster_min_size: 5
        size_of_cluster_rejection: 5
        voxel_resolution: 0.01
        check_radius: 0.1
        number_clearing_threshold: 2
        enable_near_blocked_protection: False
        number_points_considered_as_blocked: 5
        observation_sources: left_depth_cam right_depth_cam
        left_depth_cam:
          sensor_frame: camera_left_link
          topic: /depth2pc_left_node/point_cloud_from_depth
          expected_update_rate: 0.3
          FOV_W: 1.3962
          FOV_V: 0.9599
          min_detect_distance: 0.15
          max_detect_distance: 4.0
          min_obstacle_height: 0.2
        right_depth_cam:
          sensor_frame: camera_right_link
          topic: /depth2pc_right_node/point_cloud_from_depth
          expected_update_rate: 0.3
          FOV_W: 1.3962
          FOV_V: 0.9599
          min_detect_distance: 0.15
          max_detect_distance: 4.0
          min_obstacle_height: 0.2   
      inflation_layer:
        plugin: "nav2_costmap_2d::InflationLayer"
        cost_scaling_factor: 3.0
        inflation_radius: 0.8
      always_send_full_costmap: True

map_server:
  ros__parameters:
    use_sim_time: False

planner_server:
  ros__parameters:
    expected_planner_frequency: 20.0
    use_sim_time: False
    planner_plugins: ["GridBased"]
    GridBased:
      plugin: "nav2_navfn_planner/NavfnPlanner"
      tolerance: 0.5
      use_astar: false
      allow_unknown: true
